src/coreclr/tools/Common/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs

```diff
@@ -173,7 +173,9 @@ protected virtual ComputedInstanceFieldLayout ComputeInstanceFieldLayout(Metadat
             {
                 return ComputeExplicitFieldLayout(type, numInstanceFields);
             }
-            else if (type.IsSequentialLayout || type.IsEnum || type.Context.Target.Abi == TargetAbi.CppCodegen)
+            // Sequential layout has to be respected for blittable types only. We use approximation and respect it for
+            // all types without GC references (ie C# unmanaged types). Universal canonical instances might be blittable.
+            else if (type.IsSequentialLayout && (type.IsCanonicalSubtype(CanonicalFormKind.Universal) || !type.ContainsGCPointers))
```

Comment:
The field offsets in universal type have to match the field offsets of all non-universal types; or the field offset in universal type has to be undermined. This condition violates that invariant. 

I am wondering whether it would be better to disable the tests for universal layout instead of adding this change that makes the tests work, but does not actually make the code work correctly.

Reply:
We can key off SupportsUniversalCanon to do the old thing. I think that's what we'll have to do if we ever activate universal canon here.